packages/components/breadcrumbs/breadcrumbs.ts

```diff
@@ -0,0 +1,224 @@
+import { NgTemplateOutlet } from '@angular/common';
+import {
+    AfterContentInit,
+    booleanAttribute,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChild,
+    ContentChildren,
+    DestroyRef,
+    Directive,
+    forwardRef,
+    inject,
+    InjectionToken,
+    Input,
+    OnInit,
+    Provider,
+    QueryList,
+    TemplateRef,
+    ViewEncapsulation
+} from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { RouterLink } from '@angular/router';
+import { KbqButton, KbqButtonModule, KbqButtonStyles } from '@koobiq/components/button';
+import { KbqComponentColors, KbqDefaultSizes, PopUpPlacements } from '@koobiq/components/core';
+import { KbqDropdownModule } from '@koobiq/components/dropdown';
+import { KbqIconModule } from '@koobiq/components/icon';
+import { KbqTitleModule } from '@koobiq/components/title';
+import { RdxRovingFocusGroupDirective, RdxRovingFocusItemDirective } from '@radix-ng/primitives/roving-focus';
+
+export type KbqBreadcrumbsConfiguration = {
+    /**
+     * Specifies the maximum number of breadcrumb items to display.
+     * - If a number is provided, only that many items will be shown.
+     * - If `null`, no limit is applied, and all breadcrumb items are displayed.
+     */
+    max: number | null;
+    size: KbqDefaultSizes;
+};
+
+const KBQ_BREADCRUMBS_DEFAULT_CONFIGURATION: KbqBreadcrumbsConfiguration = {
+    max: 4,
+    size: 'normal'
+};
+
+/** Breadcrumbs options global configuration provider. */
+export const KBQ_BREADCRUMBS_CONFIGURATION = new InjectionToken<KbqBreadcrumbsConfiguration>(
+    'KBQ_BREADCRUMBS_CONFIGURATION',
+    { factory: () => KBQ_BREADCRUMBS_DEFAULT_CONFIGURATION }
+);
+
+/** Utility provider for `KBQ_BREADCRUMBS_CONFIGURATION`. */
+export const provideKbqBreadcrumbsConfiguration = (configuration: KbqBreadcrumbsConfiguration): Provider => ({
+    provide: KBQ_BREADCRUMBS_CONFIGURATION,
+    useValue: configuration
+});
+
+@Directive({
+    selector: '[kbqBreadcrumbsSeparator]',
+    standalone: true,
+    host: {
+        class: 'kbq-breadcrumbs-separator'
+    }
+})
+export class KbqBreadcrumbsSeparator {
+    readonly templateRef = inject(TemplateRef);
+}
+
+/**
+ * Directive to style and configure buttons used as breadcrumb items.
+ * - Inherits focus management behavior from `RdxRovingFocusItemDirective`.
+ * - Optionally injects `KbqButton` to customize its style for breadcrumb usage.
+ */
+@Directive({
+    standalone: true,
+    selector: '[kbq-button][kbqBreadcrumb]',
+    host: { class: 'kbq-breadcrumb-item' },
+    hostDirectives: [RdxRovingFocusItemDirective]
+})
+export class KbqBreadcrumbButton implements OnInit {
+    private readonly button = inject(KbqButton, { optional: true, host: true });
```

Comment:
у тебя в селекторе `[kbq-button]`, поэтому не может быть `optional`
```suggestion
    private readonly button = inject(KbqButton, { host: true });
```

Reply:
кажется что в твоем случае нужно использовать не `host` а `self`

Reply:
@artembelik optional на случай если есть атрибут, но по какой-то причине KbqButton не заимпортили, чтобы приложение не прилегло только из-за этого

Reply:
а это не сломает логику компонента?

Reply:
нет, там ведь как раз идет проверка `if (button)` далее, т.к. эта директива только устанавливает определенные свойства для `kbqButton`